hpx/components/containers/unordered/partition_unordered_map_component.hpp

```diff
@@ -0,0 +1,601 @@
+
+            data_type& m_;
+            typename data_type::iterator& it_;
+        };
+
+        T get_value(Key const& key, bool erase)
+        {
+            typename data_type::iterator it = partition_unordered_map_.find(key);
+            if (it == partition_unordered_map_.end())
+            {
+                HPX_THROW_EXCEPTION(bad_parameter,
+                    "partition_unordered_map::get_value",
+                    "unable to find requested key in this partition of the "
+                    "unordered_map");
+            }
+
+            if (!erase)
+                return it->second;
+
+            erase_on_exit t(partition_unordered_map_, it);
+            return it->second;
+        }
+
+        /// Return the element at the position \a pos in the partition_unordered_map
+        /// container.
+        ///
+        /// \param pos Positions of the elements in the partition_unordered_map
+        ///
+        /// \return Return the values of the elements at position represented
+        ///         by \a pos.
+        ///
+        std::vector<T> get_values(std::vector<Key> const& keys)
+        {
+            std::vector<T> result;
+            result.reserve(keys.size());
+
+            for (std::size_t i = 0; i != keys.size(); ++i)
+                result.push_back(partition_unordered_map_[keys[i]]);
+
+            return result;
+        }
+
+        ///////////////////////////////////////////////////////////////////////
+        // Modifiers API's in server class
+        ///////////////////////////////////////////////////////////////////////
+
+        /// Copy the value of \a val in the element at position \a pos in the
+        /// partition_unordered_map container.
+        ///
+        /// \param pos   Position of the element in the partition_unordered_map
+        ///
+        /// \param val   The value to be copied
+        ///
+        void set_value(Key const& pos, T const& val)
+        {
+            partition_unordered_map_[pos] = val;
+        }
+
+        /// Copy the value of \a val for the elements at positions \a pos in
+        /// the partition_unordered_map container.
+        ///
+        /// \param pos   Positions of the elements in the partition_unordered_map
+        ///
+        /// \param val   The value to be copied
+        ///
+        void set_values(std::vector<Key> const& keys,
+            std::vector<T> const& val)
+        {
+            HPX_ASSERT(keys.size() == val.size());
+            HPX_ASSERT(keys.size() <= partition_unordered_map_.size());
+
+            for (std::size_t i = 0; i != keys.size(); ++i)
+                partition_unordered_map_[keys[i]] = val[i];
+        }
+
+        /// Remove all elements from the vector leaving the
+        /// partition_unordered_map with size 0.
+        ///
+        void clear()
+        {
+            partition_unordered_map_.clear();
+        }
+
+        /// Erase the given element
+        std::size_t erase(Key const& key)
+        {
```

Comment:
The function returns the number of deleted elements. No need to check whether the key exists in the container.